pycbc/sensitivity.py

```diff
@@ -164,49 +168,43 @@ def volume_montecarlo(found_d, missed_d, found_mchirp, missed_mchirp,
         if max_param is not None:
             max_distance = max_param
         else:
-            # if no max distance given, use max distance actually injected
+            # If no max distance given, use max distance actually injected
             max_distance = max(found_d.max(), missed_d.max())
     else:
         raise NotImplementedError("%s is not a recognized parameter"
                                   % limits_param)
 
-    # volume of sphere
+    # Calculate volume of sphere out to maximum distance
     montecarlo_vtot = (4. / 3.) * numpy.pi * max_distance**3.
 
-    # arrays of weights for the MC integral
+    # Calculate weights based on distribution parameters
     if distribution_param == 'distance':
-        found_weights = found_d ** d_power
-        missed_weights = missed_d ** d_power
+        found_weights = numpy.power(found_d, d_power)
+        missed_weights = numpy.power(missed_d, d_power)
     elif distribution_param == 'chirp_distance':
-        # weight by a power of mchirp to rescale injection density to the
-        # target mass distribution
-        found_weights = found_d ** d_power * \
-                        found_mchirp ** mchirp_power
-        missed_weights = missed_d ** d_power * \
-                         missed_mchirp ** mchirp_power
-    else:
-        raise NotImplementedError("%s is not a recognized distance parameter"
-                                  % distribution_param)
-
-    all_weights = numpy.concatenate((found_weights, missed_weights))
-
-    # measured weighted efficiency is w_i for a found inj and 0 for missed
-    # MC integral is volume of sphere * (sum of found weights)/(sum of all weights)
-    # over injections covering the sphere
-    mc_weight_samples = numpy.concatenate((found_weights, 0 * missed_weights))
-    mc_sum = sum(mc_weight_samples)
+        found_weights = (numpy.power(found_d, d_power) * 
+                        numpy.power(found_mchirp, mchirp_power))
+        missed_weights = (numpy.power(missed_d, d_power) * 
+                         numpy.power(missed_mchirp, mchirp_power))
+
```

Comment:
The final `else` statement with NotImplementedError has been removed.  I don't see why.  We want the code to raise prompt and informative errors ..